### DISCUSSION TITLE
Displaying External Images Ingame

### DISCUSSION THREAD

**Discussion by V0idcake** [vrchat-community/osc#4] [Ideas]
A external program could theoretically convert an image to RGB values, then write to a collection of materials or some kind of specialized shader to recreate any image on a surface present on the avatar.

This could be used for modifying surface materials in realtime, doing a rudimentary displayshare or even just showing off your favorite images.

**Comment by DaryeDev** [comment]
Through OSC we can send text, iirc, so you could (maybe) convert this image to base64 and decode it on the avatar. I don't know if this could be possible on VRChat end, but if you try it, please let me know!

**Reply by knah** [reply]
OSC also has "arbitrary byte blob" data type, so technical details are clearly not an obstacle here. Transferring images over VRC's network, however, might be unacceptably heavy.
